site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### TFRecord File"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "y1llOTwWFzmR",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "TFRecord files are a simple format to store a sequence of binary blobs. In this case the fact that these are read linearly off disk, which may give some performance boost.\n",
+        "\n",
+        "It also gives you a portable data file that is apropriate, for example, for streaming over a network.\n",
+        "\n",
+        "First, build a `TFRecord` file from the raw image data:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "EqtARqKuHQLu",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "image_ds = tf.data.Dataset.from_tensor_slices(all_image_paths).map(tf.read_file)\n",
+        "tfrec = tf.data.experimental.TFRecordWriter('images.tfrec')\n",
+        "tfrec.write(image_ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "flR2GXWFKcO1",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Next build a dataset that reads from the TFRecord file, and decodes/reformats the images using the `preprocess_image` function we defined earlier."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "j9PVUL2SFufn",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "image_ds = tf.data.TFRecordDataset('images.tfrec').map(preprocess_image, num_parallel_calls=8)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "cRp1eZDRKzyN",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Zip that with the labels dataset we defined earlier, to get the expected `(image,label)` pairs."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "7XI_nDU2KuhS",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "ds = tf.data.Dataset.zip((image_ds, label_ds))\n",
+        "ds = ds.repeat().shuffle(buffer_size=4000)\n",
+        "ds=ds.batch(BATCH_SIZE).prefetch(1)\n",
+        "ds"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "3ReSapoPK22E",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "timeit(ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "wb7VyoKNOMms",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "This is slower than the `cache` version because we have not cached the preprocessing."
```

Comment:
This is kind of an anticlimax! Any chance of a "However, ..."?

Reply:
Sure.

I've split the TFRecord section into:

1. "Raw image data", where I dump the jpeg data directly to the TFRecord file.
2. "Serialized Tensors", where I decode and resize the images and `serialize_tensor` them into the TFRecord file.